src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+
+    private ProgrammingExercise exercise;
+
+    private String currentLocalFileName = "currentFileName";
+
+    private String currentLocalFileContent = "testContent";
+
+    private String newLocalFileName = "newFileName";
+
+    LocalRepository testRepo = new LocalRepository();
+
+    @BeforeEach
+    public void setup() throws Exception {
+        database.addUsers(0, 0, 1);
+        course = database.addEmptyCourse();
+        exercise = ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course);
+        testRepo.configureRepos("testLocalRepo", "testOriginRepo");
+
+        // add file to the repository folder
+        Path filePath = Paths.get(testRepo.localRepoFile + "/" + currentLocalFileName);
+        var file = Files.createFile(filePath).toFile();
+        // write content to the created file
+        FileUtils.write(file, currentLocalFileContent);
+
+        var testRepoUrl = new GitUtilService.MockFileRepositoryUrl(testRepo.originRepoFile);
+        exercise.setTestRepositoryUrl(testRepoUrl.toString());
+        doReturn(gitService.getRepositoryByLocalPath(testRepo.localRepoFile.toPath())).when(gitService).getOrCheckoutRepository(testRepoUrl.getURL(), true);
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        database.resetDatabase();
+        reset(gitService);
+        LocalRepository.resetLocalRepo(testRepo);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetFiles() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        var files = request.getMap(testRepoBaseUrl + exercise.getId() + "/files", HttpStatus.OK, String.class, FileType.class);
+        assertThat(files).isNotEmpty();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetFile() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("file", currentLocalFileName);
+        var file = request.get(testRepoBaseUrl + exercise.getId() + "/file", HttpStatus.OK, byte[].class, params);
+        assertThat(file).isNotEmpty();
+        assertThat(new String(file)).isEqualTo(currentLocalFileContent);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldCreateFile() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        LinkedMultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFile"))).isFalse();
+        params.add("file", "newFile");
+        request.postWithoutResponseBody(testRepoBaseUrl + exercise.getId() + "/file", HttpStatus.OK, params);
+        assertThat(Files.exists(Paths.get(testRepo.localRepoFile + "/newFile"))).isTrue();
```

Comment:
Maybe use `File.separator` instead of `/`.

Reply:
I tested it with the File.separator, it worked but I think the readability decreases with the usage of the File.separator. Therefore, i would keep the current implementation.